src/groovy/com/recomdata/security/ActiveDirectoryLdapAuthenticationExtension.groovy

```diff
@@ -0,0 +1,53 @@
+package com.recomdata.security
+import grails.util.Holders
+import org.aspectj.lang.ProceedingJoinPoint
+import org.aspectj.lang.annotation.Around
+import org.aspectj.lang.annotation.Aspect
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+
+import javax.annotation.PostConstruct
+/**
+ * Date: 06.07.2015
+ * Time: 15:05
+ */
+@Aspect
+class ActiveDirectoryLdapAuthenticationExtension {
+    boolean lowerCaseName = false
+    boolean onlyDomainNames = false
+    String domain
+
+    @PostConstruct
+    def configure() {
+        def ldapConf = Holders.grailsApplication.config.org.transmart.security.ldap
+        if (ldapConf.ad.lowerCaseName) {
+            lowerCaseName = ldapConf.ad.lowerCaseName
+        }
+        if (ldapConf.ad.onlyDomainNames) {
+            onlyDomainNames = ldapConf.ad.onlyDomainNames
+        }
+        domain = ldapConf.ad.domain
+    }
+
+    protected static Authentication toLowerCaseAuthentication(Authentication auth) {
+        if (auth.name == auth.name.toLowerCase()) {
+            return auth;
+        }
+        Authentication res = new UsernamePasswordAuthenticationToken(auth.name.toLowerCase(),
+                auth.credentials, auth.authorities)
+        res.details = auth.details
+        return res
+    }
+
+    @Around("bean(adAuthenticationProvider) && execution(* authenticate(..))")
+    def authenticate(ProceedingJoinPoint point) {
+        Authentication auth = point.args[0]
+        if (lowerCaseName) {
+            auth = toLowerCaseAuthentication(auth)
```

Comment:
You're not passing the new `Authentication` object that `toLowerCaseAuthentication()` may return to the decorated method. Is that intentional?